compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -517,6 +521,7 @@ impl ModCollector<'_> {
             for item in &mut trait_definition.items {
                 if let TraitItem::Function { generics, where_clause, .. } = &mut item.item {
                     desugar_generic_trait_bounds(generics, where_clause);
+                    reorder_where_clause(where_clause);
```

Comment:
I see a lot of these checks - would it be easier to add a constructor for where clauses instead and do this once there?

Reply:
I thought about it but `desugar_generic_trait_bounds` will transform a line like `fn foo<T: Trait>(x: T) where <T as Foo>::Bar: Baz` to `fn foo<T>(x: T) where <T as Foo>::Bar: Baz, T: Trait`. That is, we don't have the entire where clause when it's parsed.

I could inline `reorder_where_clause` inside `desugar_generic_trait_bounds`, though, it's always called right after that. Or maybe have another method that does both things. What do you think?